### ISSUE TITLE
Amber nomenclature in "protein" selection

### ISSUE THREAD

**Issue by sobuelow** [MDAnalysis/mdanalysis#1702]
### Expected behaviour

- load a pdb with Amber force-field specific residue naming (e.g. CGLY for c-terminal glycine residue)
- selection of "protein" atoms should include the CGLY protein

### Actual behaviour

- pdb including CGLY is correctly loaded into the universe but not in the atom group when selecting for "protein"

Testfile (rename to test.pdb):
[test.txt](https://github.com/MDAnalysis/mdanalysis/files/1458193/test.txt)

### Code to reproduce the behaviour

``` python
import MDAnalysis as mda

u = mda.Universe('test.pdb')
print(u.residues.resnames)
u_prot = u.select_atoms("protein")
print(u_prot.residues.resnames)
```

**Comment by tylerjereddy** [comment]
@sobuelow Thanks for this report. Indeed, this very limitation is [described in the source code](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/core/selection.py#L829) and `CGLY` is even mentioned as an example in a comment there.

It looks like there may also be N-term naming conventions for Amber as well. The solution is likely to hard code the Amber N- and C-terminal residue names into the above location in the code base, and we'd probably want a unit test that verifies that we can handle these esoteric residue names. We can probably guide you through this if you want to take a shot at it.

Off the top of my head, one might be able to save a bit of time by using i.e., a list comprehension to iterate over the standard 20 amino acids and generate the new residue names either preceded by `N` or `C`.

**Comment by sobuelow** [comment]
Thanks for the answer! From inspection of the amber99 aminoacids.rtp, the following residues seem not included in the source code:
'NALA', 'NGLY', 'NSER', 'NTHR', 'NLEU', 'NILE', 'NVAL', 'NASN', 'NGLN', 'NARG', 'NHID', 'NHIE', 'NHIP', 'NTRP', 'NPHE', 'NTYR', 'NGLU', 'NASP', 'NLYS', 'NPRO', 'NCYS', 'NCYX', 'NMET', 'CALA', 'CGLY', 'CSER', 'CTHR', 'CLEU', 'CILE', 'CVAL', 'CASF', 'CASN', 'CGLN', 'CARG', 'CHID', 'CHIE', 'CHIP', 'CTRP', 'CPHE', 'CTYR', 'CGLU', 'CASP', 'CLYS', 'CPRO', 'CCYS', 'CCYX', 'CMET', 'CME', 'ASF', 

I'm happy to help in testing if needed (have very little experience, though).

**Comment by richardjgowers** [comment]
@sobuelow does the code below fix this problem?

```python
import MDAnalysis as mda
import numpy as np

new_prot = np.array(['NALA', 'NGLY', 'NSER', 'NTHR', 'NLEU', 'NILE', 'NVAL', 'NASN', 'NGLN', 'NARG', 'NHID', 'NHIE', 'NHIP', 'NTRP', 'NPHE', 'NTYR', 'NGLU', 'NASP', 'NLYS', 'NPRO', 'NCYS', 'NCYX', 'NMET', 'CALA', 'CGLY', 'CSER', 'CTHR', 'CLEU', 'CILE', 'CVAL', 'CASF', 'CASN', 'CGLN', 'CARG', 'CHID', 'CHIE', 'CHIP', 'CTRP', 'CPHE', 'CTYR', 'CGLU', 'CASP', 'CLYS', 'CPRO', 'CCYS', 'CCYX', 'CMET', 'CME', 'ASF'], dtype=object)

old_prot = mda.core.selections.ProteinSelection.prot_res

mda.core.selections.ProteinSelection.prot_res = np.concatenate([old_prot, new_prot])

u = mda.Universe('test.pdb')
print(u.residues.resnames)
u_prot = u.select_atoms("protein")
print(u_prot.residues.resnames)
```

**Comment by sobuelow** [comment]
@richardjgowers It worked when replacing  `...selections...` with `...selection...`

``` 
new_prot = np.array(['NALA', 'NGLY', 'NSER', 'NTHR', 'NLEU', 'NILE', 'NVAL', 'NASN', 'NGLN', 'NARG', 'NHID', 'NHIE', 'NHIP', 'NTRP', 'NPHE', 'NTYR', 'NGLU', 'NASP', 'NLYS', 'NPRO', 'NCYS', 'NCYX', 'NMET', 'CALA', 'CGLY', 'CSER', 'CTHR', 'CLEU', 'CILE', 'CVAL', 'CASF', 'CASN', 'CGLN', 'CARG', 'CHID', 'CHIE', 'CHIP', 'CTRP', 'CPHE', 'CTYR', 'CGLU', 'CASP', 'CLYS', 'CPRO', 'CCYS', 'CCYX', 'CMET', 'CME', 'ASF'], dtype=object)

old_prot = mda.core.selection.ProteinSelection.prot_res

mda.core.selection.ProteinSelection.prot_res = np.concatenate([old_prot, new_prot])

u = mda.Universe('test.pdb')
print(u.residues.resnames)
u_prot = u.select_atoms("protein")
print(u_prot.residues.resnames)```

**Comment by kain88-de** [comment]
@sobuelow do you want to prepare a PR that fixes the problem yourself? We can show you how.

**Comment by sobuelow** [comment]
Sure.

**Comment by kain88-de** [comment]
The array with residues names is defined in [core/selection.py](https://github.com/MDAnalysis/mdanalysis/blob/90f281dadc37d0305a681971d4f77db9062bbd21/package/MDAnalysis/core/selection.py#L837). You can add to the list of residue names from the itp file with a comment like the others.  

Github has [help](https://help.github.com/articles/creating-a-pull-request/) how to create a PR and fork MDAnalysis to push your changes to github (you aren't allowed to push to the MDAnalysis repo by default).  I can help you on monday at work.
